keepalived/check/check_dns.c

```diff
@@ -388,6 +388,26 @@ dns_dump(void *data)
 	log_message(LOG_INFO, "   Name = %s", dns_check->name);
 }
 
+static int
```

Comment:
I prefer type bool, return true for same (currently 0), and false if different and change goto err to return false